src/app/core/models/rundown.ts

```diff
@@ -26,40 +26,63 @@ export class Rundown {
     rundown.infinitePieces.forEach(piece => this.infinitePieces.set(piece.layer, piece))
   }
 
-  public activate(rundownCursor: RundownCursor): void {
+  public activate(rundownCursor: RundownCursor, timestamp: number): void {
+    this.reset()
     this.isActive = true
     const segment: Segment | undefined = this.segments.find(segment => segment.id === rundownCursor.segmentId)
     if (!segment) {
       // Handle unable to activate
       return
     }
-    segment.putOnAir(rundownCursor.partId)
+    segment.putOnAir(rundownCursor.partId, timestamp)
   }
 
-  public deactivate(): void {
+  private reset(): void {
+    this.segments.forEach(segment => segment.reset())
+  }
+
+  public deactivate(timestamp: number): void {
     this.isActive = false
-    this.takeAllSegmentsOffAir()
+    this.takeAllSegmentsOffAir(timestamp)
     this.infinitePieces = new Map()
   }
 
-  private takeAllSegmentsOffAir(): void {
-    this.segments.forEach(segment => segment.takeOffAir())
+  private takeAllSegmentsOffAir(timestamp: number): void {
+    this.segments.forEach(segment => segment.takeOffAir(timestamp))
     this.segments.find(segment => segment.isNext)?.removeAsNextSegment()
   }
 
-  public takeNext(rundownCursor: RundownCursor): void {
-    this.takeAllSegmentsOffAir()
+  public takeNext(rundownCursor: RundownCursor, timestamp: number): void {
+    this.takeOnAirSegmentOffAirIfIsNextSegment(rundownCursor.segmentId, timestamp)
     const segmentToComeOnAir: Segment | undefined = this.segments.find(segment => segment.id === rundownCursor.segmentId)
     if (!segmentToComeOnAir) {
       // TODO: Handle no segment
       return
     }
-    segmentToComeOnAir.putOnAir(rundownCursor.partId)
+    segmentToComeOnAir.putOnAir(rundownCursor.partId, timestamp)
+  }
+
+  private takeOnAirSegmentOffAirIfIsNextSegment(nextSegmentId: string, timestamp: number): void {
+    const activeSegment = this.segments.find(segment => segment.isOnAir)
+    if (!activeSegment || activeSegment.id === nextSegmentId) {
+      return
+    }
+    activeSegment.takeOffAir(timestamp)
   }
```

Comment:
Something about what this method does compared to its name doesn't line up in my head.
It says it takes on-air segments off-air, if it is the next segment. However if it is the next segment it shouldn't be taken off-air, as the method correctly doesn't.

Am i misunderstanding something, or is the name missing a `Not`, i.e the name actually being `takeOnAirSegmentOffAirIfIsNotNextSegment`. 
Thats a rather long name for a pretty small method. 
Perhabs a complely new name need to be found, or the method refactored slightly?

Reply:
Oh and also adding a type to `activeSegment` would be nice. 
It is after all checked for `undefined`, due to comming from a `.find`, which can return `undefined`.

Reply:
Good observation. You are correct. I have changed the method name to `takeOnAirSegmentOffAirIfNotSetAsNext`.